spec/content/automate/ManageIQ/Cloud/VM/Provisioning/Naming.class/__methods__/vmname_spec.rb

```diff
@@ -44,12 +44,36 @@
       expect(service.object['vmname']).to eq('abcpro$n{3}')
     end
 
-    it "provisions single vm" do
+    it "provisions single vm without name from dialog" do
       provision.update!(:options => {:number_of_vms => 1})
 
       described_class.new(service).main
 
       expect(service.object['vmname']).to eq('abc$n{3}')
     end
+
+    it "provisions single vm with name from dialog" do
+      provision.update!(:options => {:number_of_vms => 1, :vm_name => "jay"})
```

Comment:
Following the vanity precedent set by "drew" :smile:

Reply:
Can you add a test for the `suffix` method?

Reply:
Or maybe `derived_name` if `suffix` is supposed to be private?

Reply:
Actually, I see.. vmname comes from `derived_name`

Reply:
The `provisions single vm with name from dialog` isn't actually affected by the fix - I threw it in for completeness while I was in here - so it will pass with/without.

I did verify locally that both new `...when number of vms is nil` test will fail without d67e8c7.

Reply:
Thanks... I'm very naive in this area so want to make sure the new tests would protect us in the future.

Reply:
No problem, thanks for all of your help!